syft/execution/plan.py

```diff
@@ -214,7 +215,7 @@ def add_placeholder(self, tensor, node_type=None):
             tensor: the tensor to replace with a placeholder
             node_type: Should be "input" or "output", used to tag like this: #<type>-*
         """
-        if tensor.id not in self.placeholders.keys():
+        if tensor.id not in self.placeholders:
```

Comment:
I think we still want `self.placeholders` indexed by integers and pointing to `Placeholder` objects.

Reply:
(Then we'd be able to fetch the right Placeholder using the value contained in an ObjectId.)

Reply:
Yep, I didn't change anything here. I just removed the `.keys()` because it's not needed: when you check if `x in dict`, Python will look at the keys of `dict`. But I can put it back if it's clearer

Reply:
Ah, makes sense.